myko_api_client/websocket_device.py

```diff
@@ -0,0 +1,90 @@
+import asyncio
+import logging
+import threading
+import traceback
+import websockets
+import json
+from time import time
+from myko_api_client.utils import retrieve_jwt_key_from_server
+from myko_api_client.constants import (
+    WEBSOCKET_ERROR_THRESHOLD_SECONDS,
+    WEBSOCKET_MAX_CONNECTION_RETRIES,
+    WEBSOCKET_WAIT_BEFORE_RETRY_SECONDS)
+
+
+class WebsocketMessageReceiver:
+    def __init__(self, rest_client):
+        self.client = rest_client
+        self.command_response_buffer = []
+
+    def _handle_event_message(self, message):
+        if "event" in message and message.get("event") == "offers_bids_response":
+            self.client.on_bids_offers_response(message)
+
+    def received_message(self, message):
+        try:
+            self._handle_event_message(message)
+        except Exception as e:
+            logging.error(f"Error while processing incoming message {message}. Exception {e}.\n"
+                          f"{traceback.format_exc()}")
+
+
+class WebsocketAsyncConnection:
```

Comment:
[Same applies](https://github.com/gridsingularity/myko-api-client/pull/1/files#r630858113) to be reused from `d3a-api-client`